src/inspect_ai/_view/www/src/samples/SamplesDescriptor.mjs

```diff
@@ -143,162 +133,132 @@ export const createsSamplesDescriptor = (
+          }),
+      ),
+    ];
+    const uniqScoreTypes = [
+      ...new Set(uniqScoreValues.map((scoreValue) => typeof scoreValue)),
+    ];
 
-  // normalize to base 1
-  const maxSizes = {
-    input: Math.min(sizes[0], 300),
-    target: Math.min(sizes[1], 300),
-    answer: Math.min(sizes[2], 300),
-    limit: Math.min(sizes[3], 50),
-    id: Math.min(sizes[4], 10),
-    score: Math.min(sizes[4], 30),
-  };
-  const base =
-    maxSizes.input +
-      maxSizes.target +
-      maxSizes.answer +
-      maxSizes.limit +
-      maxSizes.id +
-      maxSizes.score || 1;
-  const messageShape = {
-    raw: {
-      input: sizes[0],
-      target: sizes[1],
-      answer: sizes[2],
-      limit: sizes[3],
-      id: sizes[4],
-      score: sizes[5],
-    },
-    normalized: {
-      input: maxSizes.input / base,
-      target: maxSizes.target / base,
-      answer: maxSizes.answer / base,
-      limit: maxSizes.limit / base,
-      id: maxSizes.id / base,
-      score: maxSizes.score / base,
-    },
+    for (const categorizer of scoreCategorizers) {
+      const scoreDescriptor = categorizer.describe(
+        uniqScoreValues,
+        uniqScoreTypes,
+      );
+      if (scoreDescriptor) {
+        scoreDescriptorMap.set(scoreLabelKey(scoreLabel), scoreDescriptor);
+        break;
+      }
+    }
+  }
+
+  /**
+   * @param {import("../Types.mjs").ScoreLabel} scoreLabel
+   * @returns {ScoreDescriptor}
+   */
+  const scoreDescriptor = (scoreLabel) => {
+    return scoreDescriptorMap.get(scoreLabelKey(scoreLabel));
   };
 
-  const scoreRendered = (sample) => {
-    const score = scoreValue(sample);
+  /**
+   * @param {import("../api/Types.mjs").SampleInterface} sample
+   * @param {import("../Types.mjs").ScoreLabel} scoreLabel
+   * @returns {any}
+   */
+  const scoreRendered = (sample, scoreLabel) => {
+    const descriptor = scoreDescriptor(scoreLabel);
+    const score = scoreValue(sample, scoreLabel);
     if (score === null || score === "undefined") {
       return "null";
-    } else if (scoreDescriptor.render) {
-      return scoreDescriptor.render(score);
+    } else if (descriptor.render) {
+      return descriptor.render(score);
     } else {
       return score;
     }
   };
 
-  const scorerDescriptor = (sample, scorer) => {
+  /**
+   * @param {import("../api/Types.mjs").SampleInterface} sample
+   * @param {import("../Types.mjs").ScoreLabel} scoreLabel
+   * @returns {ScorerDescriptor}
+   */
+  const scorerDescriptor = (sample, scoreLabel) => {
     return {
       explanation: () => {
-        return scoreExplanation(sample, scorer);
+        return scoreExplanation(sample, scoreLabel.scorer);
       },
       answer: () => {
-        return scoreAnswer(sample, scorer);
+        return scoreAnswer(sample, scoreLabel.scorer);
       },
       scores: () => {
         if (!sample || !sample.scores) {
           return [];
         }
+        const myScoreDescriptor = scoreDescriptor(scoreLabel);
```

Comment:
I'm seeing an error when I attempt to view samples from a complex scored eval (example below):

```python
from inspect_ai import Task, task
from inspect_ai.dataset import Sample
from inspect_ai.solver import TaskState, basic_agent, prompt_template
from inspect_ai.scorer import scorer, accuracy, stderr, Target, Score, match, mean
import random




template_prompt = """
Please answer the following question with a single word that best answers the question:

{prompt}
"""

@task
def complex_score():
  return Task(
        dataset=[
            Sample(input="What the most common type of ICE powered transportation in the United States?", target="car"), 
            Sample(input="What has two wheels, pedals, and people ride it?", target=["bike", "bicycle"]), 
            Sample(input="What is a small freshwater body of water called?", target="pond"),
            Sample(input="What word is spelled incorrectly in every single dictionary?", target="incorrectly"),
            Sample(input=" Everyone in the world needs it, but they usually give it without taking it. What is it?", target="advice"),
            Sample(input="This question has an intentionally wrong target. What is the coolest word?", target="jambalaya")
        ],
        solver=basic_agent(
            init=prompt_template(template=template_prompt)
        ),
        scorer=[numeric_scorer(), match(), complex_scorer()],
    )

@scorer(metrics=[accuracy(), stderr()])
def numeric_scorer():

    async def score(state: TaskState, target: Target):

        # check for correct
        answer = state.output.completion
        value = len(target.text)

        # return score
        return Score(
            value = value,
            answer=answer
        )

    return score


@scorer(metrics={"*": [mean(), stderr()], "*cool": [accuracy(), stderr()]})    
def complex_scorer():
    async def score(state: TaskState, target: Target):

        return Score(value={
            "basic": 1.0 * random.random(),
            "fancy": 0.9 * random.random(),
            "cool": 1.1 * random.random(),
            "uncool": 2.1 * random.random()
        })
    return score
```

Would you mind having a look?

Reply:
Oops. Sorry about that. Should be fixed now.